### ISSUE TITLE
Reallymine can't find key sector if it encounters bad sector?

### ISSUE THREAD

**Issue by jgan96** [andlabs/reallymine#10]
If Reallymine encounters a bad sector when searching for the key sector on a disk, it seems to return "key sector not found" instead of skipping over the bad sector. I have verified that my disk has some bad sectors in the beginning, but the key exists in mod 0x25. Any way to manually input the mod or set a flag to skip bad sectors?

**Comment by andlabs** [comment]
That's strange; I thought a bad sector would result in an I/O error, but I guess not?

I need to add options for specifying key sectors from an alternate source. I already started working on code to extract the mods directly.

**Comment by jgan96** [comment]
You are correct, I believe the issue originally was the USB to sata cable I was using wouldn't let me read all areas of the drive or something like that. I switched to a desktop machine with an internal sata port and it found the keysector fine, however I am now getting I/O error when I try to run decrypt. Is there any way around this?

**Comment by jgan96** [comment]
For now I just ddrescued the drive and it seems to be able to decrypt that one. If it completes I will post the results.

**Comment by andlabs** [comment]
What is your ddrescue log file?

**Comment by jgan96** [comment]
I believe this is it. I made the clone a couple weeks ago so not 100% sure.
[16049.zip](https://github.com/andlabs/reallymine/files/523219/16049.zip)

**Comment by andlabs** [comment]
Yep, you have quite a few bad sectors early on in the drive. We'd have to know what the partition layout is (if this isn't a single partition's image standalone) to find out which files are lost in that, unless you want to try running ddrescue again (using the logfile to tell ddrescue what it already did).
